src/test/run-make-fulldeps/c-link-to-rust-va-list-fn/checkrust.rs

```diff
@@ -68,3 +70,10 @@ pub unsafe extern "C" fn check_list_copy_0(mut ap: VaList) -> usize {
         }
     })
 }
+
+#[no_mangle]
+pub unsafe extern "C" fn check_varargs_0(_: c_int, mut ap: ...) -> usize {
+    continue_if!(ap.arg::<c_int>() == 42);
+    continue_if!(compare_c_str(ap.arg::<*const c_char>(), "Hello, World!"));
```

Comment:
Please include a couple of tests of passing structs, too.

Reply:
I'll definitely add more tests.

> Please include a couple of tests of passing structs, too. One for a struct that's the size of a machine word, and another for a struct that contains a half-dozen machine words.

We currently only support a select number of primitive types [here](https://github.com/rust-lang/rust/blob/master/src/libcore/ffi.rs#L130-L164). Arbitrary types will result in a compiler error. AFAIK we would need to manually implement a few more `va_list` types to make that a possibility. Some more testing and exploration at the very least.

> Please also include tests for passing an f64, and a struct containing an f64; platform ABIs can vary on those points.

:+1: Good point